frappe/model/utils/__init__.py

```diff
@@ -152,6 +152,47 @@ def is_single_doctype(doctype: str) -> bool:
 
 
 def simple_singledispatch(func):
+	"""
+	A decorator that implements a simplified version of single dispatch.
+
+	This decorator allows you to define a generic function that can have
+	different behaviors based on the type of its first argument. It's similar
+	to Python's functools.singledispatch, but with a simpler implementation.
+
+	Args:
+	    func (callable): The base function to be decorated.
+
+	Returns:
+	    callable: A wrapper function that implements the single dispatch logic.
+
+	The returned wrapper function has a 'register' method that can be used
+	to register type-specific implementations:
+
+	@wrapper.register(specific_type)
+	def type_specific_func(arg, ...):
+	    # Implementation for specific_type
+
+	When the wrapped function is called, it dispatches to the most specific
+	implementation based on the type of the first argument. If no matching
+	implementation is found, it falls back to the base function.
```

Comment:
That's also the difference, at the time of writing, to `functools.singledispatch`